### PR TITLE
initial pr for the sistrip control view dqm plots

### PR DESCRIPTION
**Pull request by gouskos**
This PR produces a set of DQM plots for the SiStrip subdetector. The plots currently in DQM are organized based on the "Mechanical View" [i.e. physical subdetectors: TEC+/-, TID, TOB, TIB]. This is not always very useful to spot issues on the Control side. Hence I ve put a new dir in the SiSTrip DQM directory that is organized based on the SiStrip partiions seen in the DAQ [TEC+, TEC-, TOB, TIB/TOB] The first set of plots inlcudes: S/N for each TTC partition, FECSlot vs FEC Ring and FECCratevsFECSlot

### CODE REVIEW

**Comment by boudoul** [comment]
hello @gouskos thanks for this new PR - When making a PR,  a summary of what the PR does, add , new functionality etc is always welcome  [ see the box above you can edit  , where iy is written _No description provided_ , better to provide a description instead ]

**Comment by gouskos** [comment]
OK sorry - thought it was clear from the title. 

This PR produces a set of DQM plots for the SiStrip subdetector. The plots currently in DQM are organized based on the "Mechanical View" [i.e. physical subdetectors: TEC+/-, TID, TOB, TIB]. This is not always very useful to spot issues on the Control side. Hence I ve put a new dir in the SiSTrip DQM directory that is organized based on the SiStrip partiions seen in the DAQ [TEC+, TEC-, TOB, TIB/TOB] The first set of plots inlcudes: S/N for each TTC partition, FECSlot vs FEC Ring and FECCratevsFECSlot

**Comment by boudoul** [comment]
thanks but  put this description in  see the description box above you can edit  [where it is written No description provided , edit and replace by this description  ]

**Comment by dmitrijus** [comment]
please test

**Comment by dmitrijus** [comment]
+1

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 1]
```diff
@@ -1142,6 +1196,111 @@ SiStripMonitorTrack::Det2MEs SiStripMonitorTrack::findMEs(const TrackerTopology*
...
+bool SiStripMonitorTrack::fillControlViewHistos(const edm::Event& ev, const edm::EventSetup& es) {
+
+  edm::Handle<reco::TrackCollection > tracks;
+  ev.getByToken(trackToken_, tracks);//takes the track collection
+
+  //check that tracks are valid
+  if( !tracks.isValid() )     return false;
+
+  // loop over the tracks
+  for( reco::TrackCollection::const_iterator track = tracks->begin();
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: hi @gouskos - please use
for ( const auto & track : tracks) {...
}

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 2]
```diff
@@ -1142,6 +1196,111 @@ SiStripMonitorTrack::Det2MEs SiStripMonitorTrack::findMEs(const TrackerTopology*
...
+       track != tracks->end(); ++track ) {
+
+    // loop over the rechits of this track
+    for (trackingRecHit_iterator hit = track->recHitsBegin(), ehit = track->recHitsEnd();
+         hit!=ehit; ++hit) {
+
+      uint32_t thedetid = (*hit)->rawId();
+      if ( !(SiStripDetId(thedetid).subDetector() >=3 &&  SiStripDetId(thedetid).subDetector() <=6) ) { continue; }
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: Do not use Si*DetId. In this case, DetId(thedetId).subDetector() will work ok.

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 3]
```diff
@@ -1142,6 +1196,111 @@ SiStripMonitorTrack::Det2MEs SiStripMonitorTrack::findMEs(const TrackerTopology*
...
+        sovn = SiStripClusterInfo_.signalOverNoise();
+      }
+
+      else if (!hit1D && hit2D) {
+        const SiStripCluster* SiStripCluster_ = &*(hit2D->cluster());
+        SiStripClusterInfo SiStripClusterInfo_(*SiStripCluster_,es,thedetid);
+        sovn = SiStripClusterInfo_.signalOverNoise();
+      }
+
+
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: no need for the clear()

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 4]
```diff
@@ -1142,6 +1196,111 @@ SiStripMonitorTrack::Det2MEs SiStripMonitorTrack::findMEs(const TrackerTopology*
...
+      std::vector<const FedChannelConnection *> getFedChanConnections; getFedChanConnections.clear();
+      getFedChanConnections = SiStripDetCabling_->getConnections(thedetid);
+
+      SiStripFolderOrganizer folder_organizer;
+      TString sistripsubdet = folder_organizer.getSubDetFolderAndTag(thedetid, tTopo).second;
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: please use std::string

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 5]
```diff
@@ -1142,6 +1196,111 @@ SiStripMonitorTrack::Det2MEs SiStripMonitorTrack::findMEs(const TrackerTopology*
...
+
+      // loop over the fed chan connections
+      for (unsigned int i0=0; i0<getFedChanConnections.size(); ++i0) {
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: similarly here, a range based loop should be used

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc [hunk 6]
```diff
@@ -1142,6 +1196,111 @@ SiStripMonitorTrack::Det2MEs SiStripMonitorTrack::findMEs(const TrackerTopology*
...
+        int binfeccrate = getFedChanConnections[i0]->fecCrate();
+        int binfecslot  = getFedChanConnections[i0]->fecSlot();
+        int binfecring  = getFedChanConnections[i0]->fecRing();
+        //int binccuchan  = getFedChanConnections[i0]->ccuChan(); //will be used in a new PR
+        //int binccuadd   = getFedChanConnections[i0]->ccuAddr(); //will be used in a new PR
+
+        return2DME(ClusterStoNCorr_OnTrack_FECCratevsFECSlot,binfecslot,binfeccrate,sovn);
+
+	// TIB/TID
+        if ((sistripsubdet.Contains("TIB")) || (sistripsubdet.Contains("TID"))) {
```
Comment (davidlange6) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: avoid these string comparisons. It seems you just need the set of enums the tracker has already defined for sub detectors.

**Comment by davidlange6** [comment]
-1
pending answers to comments,.

**Comment by fioriNTU** [comment]
@gouskos please reply to comments here, otherwise we will have again a delay ...

**Comment by gouskos** [comment]
Hi Francesco, I will try to do it later today.

File: DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc (hunk 3 above)
Comment (gouskos) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: I m sorry I do not understand what you are proposing to change here. What is the clear() you referring to ?

Comment (gouskos) [inline DQM/SiStripMonitorTrack/src/SiStripMonitorTrack.cc]: Sorry found it

**Comment by gouskos** [comment]
oops i did not want those commits - fixing

**Comment by kpedro88** [comment]
-1
multiple bad merges
please follow http://cms-sw.github.io/tutorial-resolve-conflicts.html

**Comment by perrotta** [comment]
-1
(to remove from our list: it needs rebase)

**Comment by gouskos** [comment]
@kpedro88 Thanks for the link. I followed the instructions precisely but unfortunately it does not seem to work. I copy paste below what I did.

```
git cms-checkout-topic -u gouskos:sistrip-conview-92v2
git rebase --onto CMSSW_9_3_X_2017-06-27-2300 CMSSW_9_2_4 sistrip-conview-92v2

First, rewinding head to replay your work on top of it...
Applying: updating for x-trigger support
Using index info to reconstruct a base tree...
M	DQMOffline/Trigger/interface/HLTDQMFilterTnPEffHists.h
M	DQMOffline/Trigger/interface/HLTDQMTagAndProbeEff.h
M	DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc
M	DQMOffline/Trigger/python/HLTEGTnPMonitor_cfi.py
<stdin>:67: trailing whitespace.
template <typename TagType,typename TagCollType,typename ProbeType=TagType,typename ProbeCollType=TagCollType> 
<stdin>:116: trailing whitespace.
template <typename TagType,typename TagCollType,typename ProbeType,typename ProbeCollType> 
<stdin>:125: trailing whitespace.
  tagToken_ = cc.consumes<TagCollType>(pset.getParameter<edm::InputTag>("tagColl")); 
<stdin>:126: trailing whitespace.
  probeToken_ = cc.consumes<ProbeCollType>(pset.getParameter<edm::InputTag>("probeColl")); 
<stdin>:151: trailing whitespace.
template <typename TagType,typename TagCollType,typename ProbeType,typename ProbeCollType> 
warning: squelched 22 whitespace errors
warning: 27 lines add whitespace errors.
Falling back to patching base and 3-way merge...
Auto-merging DQMOffline/Trigger/python/HLTEGTnPMonitor_cfi.py
CONFLICT (content): Merge conflict in DQMOffline/Trigger/python/HLTEGTnPMonitor_cfi.py
Failed to merge in the changes.
Patch failed at 0001 updating for x-trigger support
The copy of the patch that failed is found in:
   /afs/cern.ch/work/g/gouskos/private/Tracker_2015/control_view/CMSSW_9_3_X_2017-06-27-2300/src/.git/rebase-apply/patch

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
```

fixed conflicts in :
DQMOffline/Trigger/python/HLTEGTnPMonitor_cfi.py
[always keep the HEAD]


```
git add DQMOffline/Trigger/python/HLTEGTnPMonitor_cfi.py
git rebase --continue

Applying: updating for x-trigger support
No changes - did you forget to use 'git add'?
If there is nothing left to stage, chances are that something else
already introduced the same changes; you might want to skip this patch.

When you have resolved this problem, run "git rebase --continue".
If you prefer to skip this patch, run "git rebase --skip" instead.
To check out the original branch and stop rebasing, run "git rebase --abort".
```

```
git cms-checkdeps -a
Checking out these packages: 0
scram b

git push -f my-cmssw sistrip-conview-92v2
Everything up-to-date
``` 

So I have nothing to push. Could you please advice how to proceed? Thank you.

**Comment by kpedro88** [comment]
@gouskos I think the problem is that you have previously used `cms-merge-topic` (possibly more than once?). You should never do that in a pull request.

Therefore, you first need to reset your branch to its state before you did any merges:
```
git reset --hard e9d526d
```

Then, when you do the rebase, you need to use the correct base, which is the first commit *before* any of your changes:
```
git rebase --onto CMSSW_9_3_X_2017-06-27-2300 fc51b5b sistrip-conview-92v2
```

N.B. I figured out which commit hashes to use with the following specialized `git log` command:
```
git log --graph --decorate --pretty=oneline --abbrev-commit
```

**Comment by gouskos** [comment]
Hi @kpedro88. Thank you very much for your suggestions. I tried the above instructions but again when I try to push my changes I get "Everything up-to-date".

Should I close this PR and start things from scratch -> pointing to this PR ?

**Comment by kpedro88** [comment]
@gouskos did you reuse your previous working area, or did you start from scratch? You need to start from scratch, using the two commands I wrote after you do the `cms-checkout-topic`. I tried this with your branch and it worked.

**Comment by gouskos** [comment]
Hi @kpedro88 - I will try in a clean one now. Thanks a lot for the suggestion

**Comment by gouskos** [comment]
Hi @kpedro88 ; it looks much better now - thanks a lot for your help! I will check if the changes I made to address David Lange's comments are in

**Comment by boudoul** [comment]
ok it seems that you managed to update the PR , thanks (and thanks @kpedro88 for the kind help )  @gouskos  did you eventually check that the changes are addressing all comments  ?

**Comment by boudoul** [comment]
Hi @dmitrijus and @davidlange6   : @gouskos just told me that the comments should have been adressed, could you please retrigger the tests ? Many thanks.

**Comment by davidlange6** [comment]
please test

**Comment by boudoul** [comment]
Hi @dmitrijus  can you please review this PR after the updates ?

**Comment by davidlange6** [comment]
merge

**Pull request merged**